applications/RomApplication/custom_utilities/rom_residuals_utility.h

```diff
@@ -138,17 +140,183 @@ namespace Kratos
+                        }
+                        if(aux.size1() != dofs.size() || aux.size2() != mRomDofs) {
+                            aux.resize(dofs.size(), mRomDofs,false);
+                        }
+                        RomAuxiliaryUtilities::GetPhiElemental(PhiElemental, dofs, geom, MapPhi);
+                        noalias(aux) = prod(LHS_Contribution, PhiElemental);
+                        for(int d = 0; d < static_cast<int>(dofs.size()); ++d){
+                            if (dofs[d]->IsFixed()==false){
+                                row(tempAPhi,dofs[d]->EquationId()) += row(aux,d);// Add contributions to global system for free dofs.
+                            }
+                        }
+                    }
+                }
+
+                #pragma omp for nowait
+                for (int k = 0; k < static_cast<int>(nConditions); k++){
+                    auto it = cond_begin + k;
+
+                    // Detect if the element is active or not. If the user did not make any choice the condition is active by default
+                    bool condition_is_active = true;
+                    if ((it)->IsDefined(ACTIVE)) {
+                        condition_is_active = (it)->Is(ACTIVE);
+                    }
+
+                    // Calculate condition contribution
+                    if (condition_is_active) {
+                        Condition::DofsVectorType dofs;
+                        it->GetDofList(dofs, r_current_process_info);
+                        mpScheme->CalculateLHSContribution(*it, LHS_Contribution, EquationId, r_current_process_info);
+                        const auto &geom = it->GetGeometry();
+                        if(PhiElemental.size1() != dofs.size() || PhiElemental.size2() != mRomDofs) {
+                            PhiElemental.resize(dofs.size(), mRomDofs,false);
+                        }
+                        if(aux.size1() != dofs.size() || aux.size2() != mRomDofs) {
+                            aux.resize(dofs.size(), mRomDofs,false);
+                        }
+                        RomAuxiliaryUtilities::GetPhiElemental(PhiElemental, dofs, geom, MapPhi);
+                        noalias(aux) = prod(LHS_Contribution, PhiElemental);
+                        for(int d = 0; d < static_cast<int>(dofs.size()); ++d){
+                            if (dofs[d]->IsFixed()==false){
+                                row(tempAPhi,dofs[d]->EquationId()) += row(aux,d);
+                            }
+                        }
+                    }
+                }
+
+                #pragma omp critical
+                {
+                    noalias(APhi) += tempAPhi;
+                }
+
+            }
+            return APhi;
+        }
+
     protected:
         std::vector< std::string > mNodalVariablesNames;
         int mNodalDofs;
-        unsigned int mRomDofs;
+        unsigned int mRomDofs, mPetrovGalerkinRomDofs;
```

Comment:
```suggestion
        unsigned int mRomDofs
        unsigned int mPetrovGalerkinRomDofs;
```
Though your syntax is 100% valid, in Kratos it is customary to do it like this.